docs/make.jl

```diff
@@ -53,7 +53,10 @@ makedocs(
       ]
   ],
     "Contributing" => [
-      "contributing/guidelines.md"
+      "contributing/guidelines.md",
+      "Developer guide" => [
+        "contributing/developer_guide/benchmarking.md"
```

Comment:
What about making this a single page without subfolders? `contributing/devguide.md`? My impression is that it won't evolve into a huge guide so it may be unnecessary to introduce a subfolder structure. We could have a single markdown file with sections. And if it grows we split it into various markdown files.